src/index.js

```diff
@@ -1,5 +1,19 @@
+import { AppContainer } from 'react-hot-loader';
 import React from 'react';
 import ReactDOM from 'react-dom';
 import App from './App';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+const rootEl = document.getElementById('root');
+ReactDOM.render(
+  <AppContainer component={App} />,
+  rootEl
+);
+
+if (module.hot) {
+  module.hot.accept('./App', () => {
+    ReactDOM.render(
+      <AppContainer component={require('./App').default} />,
```

Comment:
@sompylasar When you use Webpack 2 for ES modules, it compiles them to live bindings which indeed get updated on hot reload.

@oreqizer You can wrap it in a component, e.g. `Root`, and then use it instead.